netmanager/src/views/pages/Download/Download.js

```diff
@@ -186,7 +252,76 @@ const Download = (props) => {
       })
       .catch((err) => console.log(err && err.response && err.response.data));
     setLoading(false);
+    setStartDate(null);
+    setEndDate(null);
+    setFileType(null);
+    setSelectedSites([]);
+    setPollutants([]);
   };
+
+  const handleUrbanBetterFormSubmit = async (e) => {
+    e.preventDefault();
+
+    setLoading(true);
+
+    let data = {
+      startDate: roundToStartOfDay(new Date(startDate).toISOString()),
+      endDate: roundToEndOfDay(new Date(endDate).toISOString()),
+      device_numbers: getValues(deviceNumbers)
+    };
+
+    await downloadUrbanBetterDataApi(data)
+      .then((resData) => {
+        if(isEmpty(resData)) {
+          return setAlertMessage("No data found!");
+        }
+
+        let filename = `airquality-data.${fileType.value}`;
+
+        if(fileType.value === "csv"){
+          const downloadUrl = window.URL.createObjectURL(new Blob([resData]));
+          const link = document.createElement("a");
+          
+          link.href = downloadUrl;
+          link.setAttribute("download", filename);
+  
+          document.body.appendChild(link);
+          link.click();
+          link.remove();
+        } else if(fileType.value === "json") {
+          Papa.parse(resData, {
+            header: true,
+            complete: response => {
+              setParsedCsvData(response.data)
+            },
+          });
+          
+          let contentType = "application/json;charset=utf-8;";
+          
+          var a = document.createElement("a");
+          a.download = filename;
+          a.href =
+            "data:" +
+            contentType +
+            "," +
+            encodeURIComponent(JSON.stringify(parsedCsvData));
+          a.target = "_blank";
+          document.body.appendChild(a);
+          a.click();
+          document.body.removeChild(a);
+        } else {
+          console.log("Something strange happened!");
```

Comment:
console logs shouldn't be pushed :)
Could use a browser alert with the message ('file type unsupported')

Reply:
@FaithDaka worked on that. Please review.